File: julia/anagram/anagram.jl
function detect_anagrams(subject, candidates)
    subject = lowercase(subject)
    to_match = sorted(subject)
    unique([s for s in candidates if lowercase(s) != subject && sorted(s) == to_match])
end

function sorted(s)
    s |> lowercase |> collect |> sort |> join
end

File: python/anagram/anagram.py
def find_anagrams(word, candidates):
    def sorted_word(s):
        return ''.join(sorted(s.lower()))
    word = word.lower()
    word_sort = sorted_word(word)
    return [w for w in candidates if w.lower() != word and sorted_word(w) == word_sort]
